galleries/examples/shapes_and_collections/line_collection.py

```diff
@@ -3,79 +3,67 @@
-ax.set_xlim(x.min(), x.max())
-ax.set_ylim(ys.min(), ys.max())
+# create a LineCollection with the half-circles
+# its properties can be set per line by passing a sequence (here used for *colors*)
+# or they can be set for all lines by passing a scalar (here used for *linewidths*)
+line_collection = LineCollection(arcs, colors=colors, linewidths=4)
+ax.add_collection(line_collection)
 
-# *colors* is sequence of rgba tuples.
-# *linestyle* is a string or dash tuple. Legal string values are
-# solid|dashed|dashdot|dotted.  The dash tuple is (offset, onoffseq) where
-# onoffseq is an even length tuple of on and off ink in points.  If linestyle
-# is omitted, 'solid' is used.
-# See `matplotlib.collections.LineCollection` for more information.
-colors = plt.rcParams['axes.prop_cycle'].by_key()['color']
-
-line_segments = LineCollection(segs, linewidths=(0.5, 1, 1.5, 2),
-                               colors=colors, linestyle='solid')
-ax.add_collection(line_segments)
-ax.set_title('Line collection with masked arrays')
 plt.show()
 
 # %%
-# In the following example, instead of passing a list of colors
-# (``colors=colors``), we pass an array of values (``array=x``) that get
-# colormapped.
+# Instead of passing a list of colors (``colors=colors``), we can use colormapping.
+# The data to be mapped are passed via the ``array`` parameter.
+
+num_arcs = 15
+theta = np.linspace(0, np.pi, 36)
+radii = np.linspace(4, 5.5, num=num_arcs)
+arcs = [np.column_stack([r * np.cos(theta), r * np.sin(theta)]) for r in radii]
+arc_numbers = np.arange(num_arcs)
 
-N = 50
-x = np.arange(N)
-ys = [x + i for i in x]  # Many sets of y to plot vs. x
-segs = [np.column_stack([x, y]) for y in ys]
+fig, ax = plt.subplots(figsize=(6.4, 3))
+# set axes limits manually because Collections do not take part in autoscaling
+ax.set_xlim(-6, 6)
+ax.set_ylim(0, 6)
+ax.set_aspect("equal")  # to make the arcs look circular
 
-fig, ax = plt.subplots()
-ax.set_xlim(np.min(x), np.max(x))
-ax.set_ylim(np.min(ys), np.max(ys))
+# create a LineCollection with the half-circles and color mapping
+line_collection = LineCollection(arcs, array=arc_numbers, cmap='rainbow')
+ax.add_collection(line_collection)
 
-line_segments = LineCollection(segs, array=x,
-                               linewidths=(0.5, 1, 1.5, 2),
-                               linestyles='solid')
-ax.add_collection(line_segments)
-axcb = fig.colorbar(line_segments)
-axcb.set_label('Line Number')
+fig.colorbar(line_collection, label='Line number')
 ax.set_title('Line Collection with mapped colors')
-plt.sci(line_segments)  # This allows interactive changing of the colormap.
-plt.show()
 
+plt.show()
 # %%
 #
 # .. admonition:: References
 #
 #    The use of the following functions, methods, classes and modules is shown
 #    in this example:
 #
-#    - `matplotlib.collections`
 #    - `matplotlib.collections.LineCollection`
-#    - `matplotlib.cm.ScalarMappable.set_array`
+#    - `matplotlib.collections.Collection.set_array`
```

Comment:
Yes not directly, but it’s the only description of the array parameter. I believe it’s better to have that link than to have no reference to the array concept at all.

Reply:
Part of my original confusion was also the name `array`. Is there any hope of renaming that to something that more clearly evokes how it will be used?

Reply:
I'm afraid renaming is not viable or at least extremely tedious. The name is inherited from `ScalarMappable` and would thus affect all classes that support colormapping, in particular also `AxesImage`.